File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/inheritance/model/InheritanceMainModel.java
package com.nerdscorner.mvplib.testapp.events.inheritance.model;

import android.os.AsyncTask;

import com.nerdscorner.mvplib.events.model.BaseEventsModel;

public class InheritanceMainModel extends BaseEventsModel {

    public void doSomethingInBackground() {
        new SomeBackgroundTask(this).execute();
    }

    private static class SomeBackgroundTask extends AsyncTask<Void, Void, Void> {
        private InheritanceMainModel model;

        SomeBackgroundTask(InheritanceMainModel model) {
            this.model = model;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            try {
                Thread.sleep(1000L);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            return null;
        }

        @Override
        protected void onPostExecute(Void aVoid) {
            super.onPostExecute(aVoid);
            model.getBus().post(new BackgroundTaskCompletedEvent());
        }
    }

    public static class BackgroundTaskCompletedEvent {
    }
}

File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/inheritance/model/InheritanceMainModel.kt
package com.nerdscorner.mvplib.testapp.events.inheritance.model

import android.os.AsyncTask

import com.nerdscorner.mvplib.events.model.BaseEventsModel

class InheritanceMainModel : BaseEventsModel() {

    fun doSomethingInBackground() {
        SomeBackgroundTask(this).execute()
    }

    private class SomeBackgroundTask internal constructor(private val model: InheritanceMainModel) :
            AsyncTask<Void, Void, Void>() {

        override fun doInBackground(vararg voids: Void): Void? {
            try {
                Thread.sleep(1000L)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }

            return null
        }

        override fun onPostExecute(aVoid: Void?) {
            super.onPostExecute(aVoid)
            model.bus.post(BackgroundTaskCompletedEvent())
        }
    }

    class BackgroundTaskCompletedEvent
}
